src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -109,5 +110,33 @@ private static Interop.libproc.rusage_info_v3 GetCurrentProcessRUsage()
         {
             return Interop.libproc.proc_pid_rusage(Environment.ProcessId);
         }
+
+        private static TimeSpan MapTime(ulong sysTime)
+        {
+            uint denom = s_timeBase_denom;
+            if (denom == default)
+            {
+                Interop.libSystem.mach_timebase_info_data_t timeBase = GetTimeBase();
+                s_timeBase_denom = denom = timeBase.denom;
+                s_timeBase_numer = timeBase.numer;
+            }
+            uint numer = s_timeBase_numer;
+
+            return new TimeSpan(Convert.ToInt64(sysTime * numer / denom / NanosecondsTo100NanosecondsFactor));
```

Comment:
Should it do the last division first, like here?

https://chromium.googlesource.com/chromium/src/+/refs/tags/58.0.3029.141/base/time/time_mac.cc#43

(I don't know what values it typically returns)

Reply:
Oh, this says it gets the 42 from 125/3. Does 125 * ticks go over ulong max?

https://stackoverflow.com/questions/69318594/how-to-get-the-true-value-of-the-system-clock-in-raw-ticks-i-e-mach-absolute-t

Reply:
Also I suppose you can store 1000 * denom 

I'm sure someone will point out that won't necessarily give the exact same result @tannergooding 🙂

Reply:
That's a good point, but I was concerned about losing some precision due to doing division first. Though I can't wrap my head around it right now. Will try to do more thorough analysis later today.

Reply:
Ok, here are my thoughts.

Typical tick values are pretty large actually. For comparison, let's consider a program that was working on a single-core CPU for a year.
```
> [TimeSpan]::FromDays(365).Ticks
315360000000000
> [long]::MaxValue
9223372036854775807
```

(here I consider `long` since `TimeSpan` takes `long` in its ctor, ot `ulong`)

This means we will overflow on a `numer` of `29248` (or if our tick value will get significantly bigger, i.e. we take 29000 years into account, or a CPU with 29k cores).

This is very far from being realistic, but it is also much closer than I expected. So, to be on the safe side, let's do the same as Chromium does: divide by our factor first, to get a hundred times more space, by losing some precision.

We may, of course, also do `value / (denom * 100) * numer` which is relatively the same, yet will lose a bit more precision as well.

Reply:
I'm also OK with asserting that numer is reasonably small ( say < 1000) as an alternative, since we suspect that will always be true.

Reply:
My current suggestion is to leave this as `sysTime / NanosecondsTo100NanosecondsFactor * numer / denom`.

I am not sure how useful an assertion would be in this code. Perhaps it'd be better to add `checked`, since we are concerned by overflow? We do not expect this code to be too performance sensitive, so `checked` might be good.

What do you think?

Reply:
Not my area, but I"m not sure an exception would be an improvement. I'd be interested in thoughts of a numeric expert like @tannergooding . I'll step aside for area owners to sign off overall...

Reply:
I would say that an explicit exception might be an improvement (especially since it would be thrown from a member property such as `TotalProcessorTime` and not on class init).

But of course, I am ready to listen to other suggestions.